### PR TITLE
NEP-51: Token Standard Requiring Optional Methods

### PR DESCRIPTION
**Pull request by anfn101**
See issue: https://github.com/neo-project/proposals/issues/59

### CODE REVIEW

File: nep-51.mediawiki [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+* Remarks: "approve" will approve the '''to''' account to transfer '''amount''' tokens from the '''originator''' acount. 
+
+===Events===
+
+====transfer====
+
+* Syntax: <code>public static event Action<byte[], byte[], BigInteger> transfer</code>
+
+* Remarks: The "transfer" event is raised after a successful execution of the "transfer" method.
+
```
Comment (localhuman) [inline nep-51.mediawiki]: Do we want an `approve` event?

Comment (anfn101) [reply nep-51.mediawiki]: For previous discussion on optional events, we can refer to the discussion on the original NEP-5 here: https://github.com/neo-project/proposals/pull/4

In defense of approve, it allows people to transfer on behalf on someone, which people may find useful.

**Comment by RavenXce** [comment]
This lacks the necessary details for the new (and also existing) methods, so we can't discuss anything. Better to base of the current NEP-5 doc https://github.com/neo-project/proposals/blob/master/nep-5.mediawiki.

**Comment by anfn101** [comment]
Have updated to be more in line with current version.

**Comment by canesin** [comment]
@RavenXce we have been recommending this implementation due to a needed change in the VM please contact me or @localhuman in discord or email, as you feel best, so we clarify the needed changes in the way you call some contracts.

We all understand that a new token standard that just reinstitute a old one is not ideal, but unfortunately a oversight let a new standard proposal at the same time change the at the time already finalized NEP5, now we have a vocal person spamming very damaging warns across social channels that will put projects following it at risk.

@anfn101 this proposal should mark NEP5 status as "replaced" so people don't fall in that mistake.

File: nep-51.mediawiki (hunk 1 above)
Comment (mwherman2000) [inline nep-51.mediawiki]: @totalvamp @afong @tyler @lllwvlvwlll @erikzhang  https://github.com/neo-project/proposals/pull/58 is non compliant with the current direction that has previously been established for NEPs. It should only contain the incremental merthods. Nothing more.  That is was agreed to.  #GangOfZion being destructive again.

@erikzhang, to you as the NEP Editor, I recommend against you accepting this NEP.

**Comment by mwherman2000** [comment]
@totalvamp @afong @tyler @lllwvlvwlll @erikzhang #58 is non compliant with the current direction that has previously been established for NEPs. It should only contain the incremental merthods. Nothing more. That is was agreed to. #GangOfZion being destructive again.

@erikzhang, to you as the NEP Editor, I recommend against you accepting this NEP.

**Comment by Edgegasm** [comment]
@erikzhang @mwherman2000 

Michael's opposition to #58 is based on a supposed community consensus for a new NEP direction & demand against retroactive changes. 

However, this _current direction_ had no community consensus; instead, Michael has justified it based on his own preferences that went largely undiscussed for 30 days. 

In addition, the only retroactive changes that have been made so far are Michael's [own handiwork](https://github.com/neo-project/proposals/commit/8c6b427e2ada2e71f92081686c4a832351e9eaa0#diff-e5a1452ea0de8fa24b9c328151e94aff), where he retroactively changed a finalized NEP and subsequently re-proposed it under his [own name](https://github.com/neo-project/proposals/pull/60/commits/e7331dc4c362c48d8e05675c033bcc38ce2809d6#diff-be542232e0c21a3a90e2065df599ca44).

I suggest this objection is ignored entirely, and NEP-51 is finalized to supersede NEP-5.

**Comment by mwherman2000** [comment]
@edgegasm You can read through the community based discussion, acceptance, and finalization of NEP-10 here: https://github.com/neo-project/proposals/issues/40

**Comment by Edgegasm** [comment]
On 12th May, you pointed out your concept of removing optional methods from NEPs going forward.

On 14th May, Erik agreed that there should not be optional methods; just new NEPs for separate methods.

On 17th May, you retroactively removed the existing methods from a finalized NEP. 

In what dimension is a conversation solely between you and Erik over the span of five days a 'community based discussion?' Do you see how _ludicrous_ that is from the perspective of someone that has been noting your own belligerence towards members of CoZ, for supposedly basing decisions on private discussions rather than community involvement? I find it to be wholly unacceptable and hypocritical.

**Comment by canesin** [comment]
Just to publicly update I talked with @RavenXce in private to clarify. The intention was not to be disrespectful to any member of the broader community but unfortunately not every issue can be discussed in the open as it could damage projects.

**Comment by saltyskip** [comment]
#61 or #60 seem to be a better solution as it does not break compatibility of existing projects, and allows us to maintain NEP-5 as the base token standard, with or without extensions

**Comment by erikzhang** [comment]
> #61 or #60 seem to be a better solution as it does not break compatibility of existing projects, and allows us to maintain NEP-5 as the base token standard, with or without extensions

Agree.

**Pull request closed without merging**